--- go/src/cmd/go/internal/envcmd/env.go	2020-06-02 04:44:34.000000000 +0900
+++ patches/src/cmd/go/internal/envcmd/env.go	2020-06-02 04:44:34.000000000 +0900
@@ -100,6 +100,11 @@
 		env = append(env, cfg.EnvVar{Name: key, Value: val})
 	}
 
+	ar := "ar"
+	if runtime.GOOS == "darwin" && (runtime.GOARCH == "arm" || runtime.GOARCH == "arm64") {
+		ar = "llvm-ar"
+	}
+	ar = envOr("AR", ar)
 	cc := cfg.DefaultCC(cfg.Goos, cfg.Goarch)
 	if env := strings.Fields(cfg.Getenv("CC")); len(env) > 0 {
 		cc = env[0]
@@ -108,7 +113,7 @@
 	if env := strings.Fields(cfg.Getenv("CXX")); len(env) > 0 {
 		cxx = env[0]
 	}
-	env = append(env, cfg.EnvVar{Name: "AR", Value: envOr("AR", "ar")})
+	env = append(env, cfg.EnvVar{Name: "AR", Value: ar})
 	env = append(env, cfg.EnvVar{Name: "CC", Value: cc})
 	env = append(env, cfg.EnvVar{Name: "CXX", Value: cxx})
 
