--- go/src/cmd/internal/sys/supported.go	2020-06-02 04:44:34.000000000 +0900
+++ patches/src/cmd/internal/sys/supported.go	2020-06-02 04:44:34.000000000 +0900
@@ -67,7 +67,7 @@
 		case "linux/amd64", "linux/arm", "linux/arm64", "linux/386", "linux/ppc64le", "linux/s390x",
 			"android/amd64", "android/arm", "android/arm64", "android/386",
 			"freebsd/amd64",
-			"darwin/amd64", "darwin/386",
+			"darwin/amd64", "darwin/arm", "darwin/arm64", "darwin/386",
 			"windows/amd64", "windows/386":
 			return true
 		}
@@ -84,7 +84,7 @@
 		case "linux/386", "linux/amd64", "linux/arm", "linux/arm64", "linux/ppc64le", "linux/s390x",
 			"android/amd64", "android/arm", "android/arm64", "android/386",
 			"freebsd/amd64",
-			"darwin/amd64",
+			"darwin/amd64", "darwin/arm", "darwin/arm64",
 			"aix/ppc64":
 			return true
 		}
@@ -101,7 +101,7 @@
 		switch platform {
 		case "linux/amd64", "linux/arm", "linux/arm64", "linux/386", "linux/s390x", "linux/ppc64le",
 			"android/amd64", "android/arm", "android/arm64", "android/386",
-			"darwin/amd64",
+			"darwin/amd64", "darwin/arm", "darwin/arm64",
 			"freebsd/amd64":
 			return true
 		}
